src/manifest/adaptation/index.ts

```diff
@@ -61,15 +67,34 @@ class Adaptation {
    * @constructor
    * @param {Object} args
    */
-  constructor(args : IAdaptationArguments) {
+  constructor(
+    args : IAdaptationArguments,
+    warning$? : Subject<Error|ICustomError>,
+    representationFilter? : CustomRepresentationFilter
+  ) {
     const nId = generateNewId();
     this.id = args.id == null ? nId : "" + args.id;
     this.type = args.type;
-    this.representations = Array.isArray(args.representations) ?
+    const argsRepresentations = filterSupportedRepresentations(
+      args.type,
       args.representations
+    );
+
+    if (argsRepresentations.length === 0 && warning$) {
+      log.warn("Incompatible codecs for adaptation", args);
+      const error =
+        new MediaError("MANIFEST_INCOMPATIBLE_CODECS_ERROR", null, false);
```

Comment:
Here it's less evident that no representations means a codec error.
We should do something about it